impl/src/error.rs

```diff
@@ -48,20 +48,21 @@ pub fn expand(
     let provide = provide.map(|provide| {
         quote! {
             fn provide<'_demand>(&'_demand self, demand: &mut ::std::any::Demand<'_demand>) {
-                 #provide
-             }
+                #provide
+            }
         }
     });
 
     let mut generics = generics.clone();
 
     if !type_params.is_empty() {
-        let generic_parameters = generics.params.iter();
+        // delegate generic parameters tokenization to `syn`
+        let (_, generic_parameters, _) = generics.split_for_impl();
```

Comment:
nit: let's make the variable naming consistent with the rest of the codebase

```suggestion
        let (_, ty_generics, _) = generics.split_for_impl();
```